### DISCUSSION TITLE
Logging a JSON string property value - "pretty print"

### DISCUSSION THREAD

**Discussion by ryanbuening** [datalust/seq-tickets#1374] [Help] [answered]
I'm logging a property value which is a JSON string. Is there a way to have this "pretty print" in Seq?

![image](https://user-images.githubusercontent.com/812974/141352003-2069616c-2f1d-4c59-b5f0-31c4925c867e.png)

**Comment by nblumhardt** [comment, marked as answer]
Hi Ryan! Directly - not really.

But, you can use `FromJson()` to parse JSON into a column of a query result, which is a bit nicer to display/navigate:

```
select FromJson(RequestBody)
from stream
where RequestBody is not null
limit 5
```

Pretty-printing/visualizing different data types is something we've got some ideas around, though :-)

**Comment by globalexport** [comment]
Hi @nblumhardt! 

Pretty-printing would be really nice. Are there updates regarding this topic?

**Reply by nblumhardt** [reply]
One slight improvement in Seq 2022.1; you can now select "Assign to $someVariable" in the green-tick drop-down menu beside the property.

Once you've done that, you can interactively inspect and manipulate the value. E.g. if the JSON property is assigned to `$requestBody0`, then run:

```sql
select FromJson($requestBody0)
```

to see the parsed/pretty-printed form.

Or assign the parsed form to another variable:

```sql
select FromJson($requestBody0) into $body
```

And then explore:

```sql
select Length($body.Items[0].Name)
```

etc.

**Reply by globalexport** [reply]
Thank you for the update! 
This is helpful to inspect one individual "object" string.

**Reply by esp0** [reply]
PS: If you are using Serilog and Deserialize the JSON beforing logging it, the data will be shown "Pretty" in the Seq GUI:
![image](https://user-images.githubusercontent.com/102811/196900440-3485a546-4e3f-4bce-952a-6ce79c3e675f.png)

**Reply by wangshuai-007** [reply]
If use **Nlog** and **NLog.Targets.Seq**
In [NLog.Targets.Seq sample](https://github.com/datalust/nlog-targets-seq/blob/b937257006e82fee90f3eb69412483178d6051e9/sample/Example/Program.cs#L21)

Show how to log object.
`Logger.Info("Current user is {@User}, height {Height:0.00}", new { Name = Environment.UserName, Tags = new[]{ 1, 2, 3 }}, 123.4567);`

So, Just `JsonConvert.DeserializeObject` the json string and use `@ParamName`
